Dockerfile

```
@@ -1,28 +1,25 @@
-FROM ubuntu:xenial
+FROM ubuntu:bionic
 
-LABEL authors='Chris Rust <chris.rust@solarwinds.com>, Dawid Śmiech <dawid.smiech@solarwinds.com>'
-
-ARG swisnap_version
-ARG swisnap_repo=swisnap
+LABEL authors='Dawid Śmiech <dawid.smiech@solarwinds.com>'
 
 USER root
 ENV DEBIAN_FRONTEND noninteractive
 
-RUN \
-  apt-get update && \
-  apt-get -y install software-properties-common && \
-  LC_ALL=C.UTF-8 add-apt-repository -y ppa:rmescandon/yq && \
-  apt-get update && \
-  apt-get -y upgrade && \
-  apt-get -y install \
-    apt-transport-https \
-    ca-certificates \
-    docker.io \
-    curl \
-    yq
+ARG swisnap_repo=swisnap
+RUN apt-get update && \
+    apt-get -y install software-properties-common && \
+    LC_ALL=C.UTF-8 add-apt-repository -y ppa:rmescandon/yq && \
+    apt-get update && \
+    apt-get -y upgrade && \
+    apt-get -y install \
+      apt-transport-https \
+      ca-certificates \
+      docker.io \
+      curl \
+      yq
```

Should we be cleaning up after apt in this layer as well?

```suggestion
      curl \
      yq \&& 
      apt-get clean && \
      rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

---

It does clean and remove curl in the next layer and apt via `rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* && \`

Do you see a benefit in doing it in both layers?

---

And files remaining in the layer when it exits the `RUN ` will be committed and contribute to the final image size

---

I agree. Each layer needs to be cleaned. Each RUN command creates a new layer and anything left inside it ends up in the final image.